tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs

```diff
@@ -1301,5 +1302,109 @@ public async Task MixedUserAndSystemAssignedManagedIdentityTestAsync()
+        public async Task SAMIFails500PermanentlyAndRetryPolicyLifeTimeIsPerRequestAsync(ManagedIdentitySource managedIdentitySource, string endpoint)
+        {
+            using (new EnvVariableContext())
+            using (var httpManager = new MockHttpManager(isManagedIdentity: true))
+            {
+                var miBuilder = ManagedIdentityApplicationBuilder.Create(ManagedIdentityId.SystemAssigned)
+                    .WithHttpManager(httpManager);
+
+                // Disable cache to avoid pollution
+                miBuilder.Config.AccessorOptions = null;
+
+                var mi = miBuilder.Build();
+
+                // Simulate permanent 500s (to trigger the maximum number of retries)
+                int NUM_500 = ManagedIdentityRequest.DEFAULT_MANAGED_IDENTITY_MAX_RETRIES + 1; // initial request + maximum number of retries (3)
+                for (int i = 0; i < NUM_500; i++)
+                {
+                    httpManager.AddManagedIdentityMockHandler(
+                        endpoint,
+                        ManagedIdentityTests.Resource,
+                        "",
+                        managedIdentitySource,
+                        statusCode: HttpStatusCode.InternalServerError);
+                }
+
+                MsalServiceException msalException = null;
+                try
+                {
+                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    msalException = ex as MsalServiceException;
+                }
+                Assert.IsNotNull(msalException);
+
+                // ensure that the first request was made and retried 3 times
+                Assert.AreEqual(LinearRetryPolicy.numRetries, ManagedIdentityRequest.DEFAULT_MANAGED_IDENTITY_MAX_RETRIES);
+                Assert.AreEqual(httpManager.QueueSize, 0);
+
+                for (int i = 0; i < NUM_500; i++)
+                {
+                    httpManager.AddManagedIdentityMockHandler(
+                        endpoint,
+                        ManagedIdentityTests.Resource,
+                        "",
+                        managedIdentitySource,
+                        statusCode: HttpStatusCode.InternalServerError);
+                }
+
+                msalException = null;
+                try
+                {
+                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    msalException = ex as MsalServiceException;
+                }
+                Assert.IsNotNull(msalException);
+
+                // ensure that the second request was made and retried 3 times
+                // (numRetries would be x2 if retry policy was NOT per request)
```

Comment:
I don't think I understand the 2nd line of this comment

Reply:
2 x (initial request + 3 retries) = 8 network requests